### PR TITLE
backend/fix/#1278-completed-shared-services-lib-and-added-customer-is…

### PR DESCRIPTION
**Pull request by VD321**
…sue-report-flow

## Type of Change

- [X] Bugfix
- [X] New feature
- [ ] Enhancement
- [X] Refactoring
- [ ] Dependency updates

## Description
Shifted issue flow for driver and dashboard to shared-service library and added the issue flow in rider-app.

### Additional Changes

- [X] This PR modifies the database schema (database migration added)
- [X] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/lib/shared-services/src/IssueManagement/Tools/UtilsTH.hs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import qualified Kernel.Beam.Lib.UtilsTH as TH
+import Kernel.Prelude hiding (catch)
+import Kernel.Types.Common as Reexport hiding (id)
+import Language.Haskell.TH
+import System.Environment (getEnv)
+import System.IO.Unsafe (unsafePerformIO)
+
+schemaName :: String
+{-# NOINLINE schemaName #-}
+schemaName = unsafePerformIO (getEnv "SCHEMA_NAME")
```
Comment (roman-bodavskiy) [inline Backend/lib/shared-services/src/IssueManagement/Tools/UtilsTH.hs:27]: Is it acceptable for us to fetch `SCHEMA_NAME` from env variables? @hkmangla 
What if we use `run-mobility-stack-dev ` command locally? Then the same `SCHEMA_NAME` will be used both for bap and bpp, right?

**Comment by tejkranthi** [comment]
@VD321 Please resolve these comments asap.

**Comment by VD321** [comment]
Will keep this pr open in case a reliable method reuse entire code is found in future. Until then will mark this pr as WIP.
Link to new pr : https://github.com/nammayatri/nammayatri/pull/3566
Above pr uses the same method as used for re-using Beckn Request table by @roman-bodavskiy

**Comment by roman-bodavskiy** [comment]
> Will keep this pr open in case a reliable method to reuse entire code is found in future.

In my example PR I reuse entire code using `BeamFlow` synonym. 
https://github.com/nammayatri/nammayatri/pull/3558/files#diff-8bc5123f6b2cfbf117462d9e6c46a89448372edfa493354fe1167684de1c5881R28

This solution not as good as we want, because it requires Orphan Instances, and potentially it can be overlapping somewhere in code. We was thinking with @vijaygupta18 about ideal solution, probably it requires changing Sequilize lib, and use Reader monad for keep `schemaName`. This is not implemented at the time

File: Backend/lib/shared-services/src/IssueManagement/Tools/UtilsTH.hs (hunk 1 above)
Comment (VD321) [reply Backend/lib/shared-services/src/IssueManagement/Tools/UtilsTH.hs]: Remove this way of fetching schemaName.

**Comment by VD321** [comment]
This PR Now uses the same method for reusing code as in @roman-bodavskiy's example pr.

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Issue.hs [hunk 2]
```diff
@@ -37,7 +41,7 @@ import "lib-dashboard" Tools.Auth hiding (BECKN_TRANSPORT)
 import "lib-dashboard" Tools.Auth.Merchant
 
 type API =
-  "issue"
+  "issueV2"
```
Comment (hkmangla) [inline Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Issue.hs:44]: not backward compatible, why to change only?

File: Backend/dev/migrations/dynamic-offer-driver-app/0275-adding-new-issue-flow-related-data.sql [hunk 3]
```diff
@@ -0,0 +1,139 @@
+ALTER TABLE atlas_driver_offer_bpp.issue_report RENAME COLUMN driver_id TO person_id;
+
+CREATE TABLE atlas_driver_offer_bpp.issue_message (
```
Comment (hkmangla) [inline Backend/dev/migrations/dynamic-offer-driver-app/0275-adding-new-issue-flow-related-data.sql:3]: These tables are already there in driver, right?

File: Backend/dev/migrations/dynamic-offer-driver-app/0275-adding-new-issue-flow-related-data.sql [hunk 4]
```diff
@@ -0,0 +1,139 @@
+ALTER TABLE atlas_driver_offer_bpp.issue_report RENAME COLUMN driver_id TO person_id;
```
Comment (hkmangla) [inline Backend/dev/migrations/dynamic-offer-driver-app/0275-adding-new-issue-flow-related-data.sql:1]: Not backward compatible.

File: Backend/lib/shared-services/src/IssueManagement/API/UI/Issue.hs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+module IssueManagement.API.UI.Issue where
+
+import qualified IssueManagement.Domain.Types.Issue.IssueReport as Domain
+import qualified IssueManagement.ProviderPlatform.Issue as Common
+import Kernel.Types.Id
+import Servant
+
+type IssueAPI =
+  Common.IssueCreateAPI
```
Comment (hkmangla) [inline Backend/lib/shared-services/src/IssueManagement/API/UI/Issue.hs:9]: Why these types are coming from Dashboard? Aren't service itself should have all the types?

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/dev/migrations/dynamic-offer-driver-app/0275-adding-new-issue-flow-related-data.sql (hunk 3 above)
Comment (VD321) [reply Backend/dev/migrations/dynamic-offer-driver-app/0275-adding-new-issue-flow-related-data.sql:3]: This is a new table.

File: Backend/lib/shared-services/src/IssueManagement/API/UI/Issue.hs (hunk 5 above)
Comment (VD321) [reply Backend/lib/shared-services/src/IssueManagement/API/UI/Issue.hs:9]: This is coming from within the library itself, however that module is only being used to reexport. We can remove it.

File: Backend/dev/migrations/dynamic-offer-driver-app/0275-adding-new-issue-flow-related-data.sql (hunk 4 above)
Comment (VD321) [reply Backend/dev/migrations/dynamic-offer-driver-app/0275-adding-new-issue-flow-related-data.sql]: Changed driver_id to maybe type, and added person_id, for customer driver_id will have null values.

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Issue.hs (hunk 2 above)
Comment (VD321) [reply Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Issue.hs]: added issueV2 for customer but kept it same for driver.

**Review by hkmangla — APPROVED** [review]

**Pull request merged**